test/integration/requests/api-versions-request.integration.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { createConnection, type Socket } from 'net';
+import { KafkaContainer, type StartedKafkaContainer } from '@testcontainers/kafka';
+import { Connection } from '../../../src/connection/connection';
+import {
+  ApiVersionsRequestV0,
+  ApiVersionsRequestV1,
+  ApiVersionsRequestV2,
+  ApiVersionsRequestV3
+} from '../../../src/protocol/requests';
+import { CompactString } from '../../../src/protocol/primitives';
+
+jest.setTimeout(120_000);
+
+const port = 9092;
+
+describe('ApiVersionsRequest', () => {
+  let socket: Socket;
+  let connection: Connection;
+  let container: StartedKafkaContainer;
+
+  beforeAll(async () => {
+    container = await new KafkaContainer('confluentinc/cp-kafka:7.3.2').withExposedPorts(port).withReuse().start();
```

Comment:
I've chosen the version based on my local setup. To test this thoroughly, I would like to run the suite against different versions and configurations, see #107.